### ISSUE TITLE
Joplin takes longer to start when clicking on app icon instead of tray icon

### ISSUE THREAD

**Issue by Candelus** [laurent22/joplin#10533]
### Operating system

Windows

### Joplin version

2.14.22

### Desktop version info

Joplin 2.14.22 (prod, win32)

Client ID: ce8ac757ee3c4bd2adfe884857f2a893
Sync Version: 3
Profile Version: 46
Keychain Supported: Yes

Revision: e579eb9

Backup: 1.4.0

### Current behaviour

Joplin takes longer to show up despite already launched the first time if opening with app icon, if opening through tray icon it's instantly loaded. I want to keep my tray clean as I'm used to having big app icons more.

Also please add a run with windows startup feature :)

Videos demonstration included

https://github.com/laurent22/joplin/assets/153918815/99cbd911-cbbb-4f38-a668-1760ce1eaa05

### Expected behaviour

Open instantly after first launch whether clicking on app icon or tray icon

**Comment by heliole** [comment]
I would also really love the feature to fast start from the app icon.

Also you can run Joplin on Windows startup with these steps:
1. Press "Win + R"
2. Write `shell:startup`
3. Put a shortcut to Joplin.exe in the folder

Now Joplin will run on Windows startup.
And if you want it to start minimized, you can check that option in Joplin settings.

**Comment by Candelus** [comment]
> I would also really love this feature.
> 
> Also you can run Joplin on Windows startup with these steps:
> 
> 1. Press "Win + R"
> 2. Write `shell:startup`
> 3. Put a shortcut to Joplin.exe in the folder
> 
> Now Joplin will run on Windows startup. And if you want it to start minimized, you can check that option in Joplin settings.

thanks for the run on windows startup solution. I'll give it a try 🙂 still need them to fix the slow opening issue though.

**Comment by Candelus** [comment]
> Hey there, it looks like there has been no activity on this issue recently. Has the issue been fixed, or does it still require the community's attention? If you require support or are requesting an enhancement or feature then please create a topic on the [Joplin forum](https://discourse.joplinapp.org/). This issue may be closed if no further activity occurs. You may comment on the issue and I will leave it open. Thank you for your contributions.

not yet

**Comment by heliole** [comment]
> > Hey there, it looks like there has been no activity on this issue recently. Has the issue been fixed, or does it still require the community's attention? If you require support or are requesting an enhancement or feature then please create a topic on the [Joplin forum](https://discourse.joplinapp.org/). This issue may be closed if no further activity occurs. You may comment on the issue and I will leave it open. Thank you for your contributions.
> 
> not yet

Did you find any workaround for this? Maybe use a script that checks if the joplin process is running and open it?

**Comment by Candelus** [comment]
> Did you find any workaround for this? Maybe use a script that checks if the joplin process is running and open it?

I haven't so I just left it on most of the time. I'm not a coder or someone working in these areas so I can't really dabble in scripts. Apologies 🥹

**Comment by Candelus** [comment]
> Hey there, it looks like there has been no activity on this issue recently. Has the issue been fixed, or does it still require the community's attention? If you require support or are requesting an enhancement or feature then please create a topic on the [Joplin forum](https://discourse.joplinapp.org/). This issue may be closed if no further activity occurs. You may comment on the issue and I will leave it open. Thank you for your contributions.

not yet

**Comment by heliole** [comment]
I can try and put something up to implement this but it would have to be reviewed. Just to get some more attention on this issue bc i'm pretty sure this issue can be fixed super easily and it would be a noticeable improvement. (Example: [Notesnook](https://github.com/streetwriters/notesnook) app is built using electron too, and it has this functionality)

**Comment by Candelus** [comment]
> super easily and it would be a noticeable improvement

I'm just leaving it opened to get the slow loading issue addressed. Regardinging the run on startup I've already done it with the shell:startup method so all good 🙂 (plus it's not the main issue anyway)

**Comment by heliole** [comment]
> > super easily and it would be a noticeable improvement
> 
> I'm just leaving it opened to get the slow loading issue addressed. Regardinging the run on startup I've already done it with the shell:startup method so all good 🙂 (plus it's not the main issue anyway)

I was addressing the slow loading issue 😅 + it's bothering me too and I bet we aren't the only ones.

**Comment by Candelus** [comment]
> > > super easily and it would be a noticeable improvement
> > 
> > 
> > I'm just leaving it opened to get the slow loading issue addressed. Regardinging the run on startup I've already done it with the shell:startup method so all good 🙂 (plus it's not the main issue anyway)
> 
> I was addressing the slow loading issue 😅 + it's bothering me too and I bet we aren't the only ones.

oh then by all means yes please!

**Comment by Candelus** [comment]
> Hey there, it looks like there has been no activity on this issue recently. Has the issue been fixed, or does it still require the community's attention? If you require support or are requesting an enhancement or feature then please create a topic on the [Joplin forum](https://discourse.joplinapp.org/). This issue may be closed if no further activity occurs. You may comment on the issue and I will leave it open. Thank you for your contributions.

not yet

**Comment by heliole** [comment]
I tried fixing the issue but it's not as simple as i thought it would be (changing few lines of code). Maybe it's just electron things.

**Comment by xsswon** [comment]
The same issue occurs: Launching Joplin by the taskbar icon is prolonged. I mostly click the taskbar icon to launch for convenience rather than the tray icon(although clicking the tray icon is fast).

**Comment by Candelus** [comment]
> The same issue occurs: Launching Joplin by the taskbar icon is prolonged. I mostly click the taskbar icon to launch for convenience rather than the tray icon(although clicking the tray icon is fast).

yes, that's what I have been waiting for them to fix since posting

**Comment by Candelus** [comment]
comment
